timemachine/fe/estimator.py

```diff
@@ -130,52 +229,31 @@ def simulate(
         barostat.seed = np.random.randint(np.iinfo(np.int32).max)
 
     intg_impl = integrator.impl()
-    baro_impl = barostat.impl(all_impls)
+    # technically we need to only pass in the nonbonded impl
+    barostat_impl = barostat.impl(all_impls)
     # context components: positions, velocities, box, integrator, energy fxns
-    ctxt = custom_ops.Context(
-        x0,
-        v0,
-        box,
-        intg_impl,
-        all_impls,
-        barostat=baro_impl,
-    )
-    base_interval = baro_impl.get_interval()
-    # Use an interval of 5 for equilibration
-    baro_impl.set_interval(5)
+    ctxt = custom_ops.Context(x0, v0, box, intg_impl, all_impls, barostat_impl)
 
     # equilibration
     equil_schedule = np.ones(equil_steps) * lamb
     ctxt.multiple_steps(equil_schedule)
 
-    baro_impl.set_interval(base_interval)
+    # (ytz): intentionally hard-coded, I'd rather the end-user *not*
+    # muck with this unless they have a good reason to.
+    barostat_impl.set_interval(25)
 
-    prod_schedule = np.ones(prod_steps) * lamb
+    full_us, xs, boxes = ctxt.multiple_steps_U(lamb, prod_steps, np.array(lambda_windows), u_interval, x_interval)
 
-    full_du_dls, xs, _ = ctxt.multiple_steps(prod_schedule, du_dl_interval, x_interval)
+    result = SimulationResult(
+        xs=xs.astype("float32"),
+        boxes=boxes.astype("float32"),
+        lambda_us=full_us,
+    )
 
-    result = SimulationResult(xs=xs.astype("float32"), du_dls=full_du_dls)
     return result
 
 
-FreeEnergyModel = namedtuple(
-    "FreeEnergyModel",
-    [
-        "unbound_potentials",
-        "client",
-        "box",
-        "x0",
-        "v0",
-        "integrator",
-        "lambda_schedule",
-        "equil_steps",
-        "prod_steps",
-        "barostat",
-    ],
-)
-
-
-def deltaG(model, sys_params) -> Tuple[float, List]:
+def deltaG_from_results(model, results, sys_params) -> Tuple[float, float, List]:
 
```

Comment:
Add docs? It might be worth mentioning model.endpoint_correct in particular, since this is specific to a system set up in a certain way.